substrate/frame/assets/src/migration/v1.rs

```diff
@@ -0,0 +1,172 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use crate::{AssetStatus, Config, DepositBalanceOf, Pallet, LOG_TARGET};
+use frame_support::{
+	pallet_prelude::*, sp_runtime::traits::Saturating, traits::OnRuntimeUpgrade, weights::Weight,
+};
+use log;
+
+#[cfg(feature = "try-runtime")]
+use crate::Vec;
+#[cfg(feature = "try-runtime")]
+use sp_runtime::TryRuntimeError;
+
+mod old {
+	use super::*;
+
+	#[derive(Decode)]
+	pub struct AssetDetails<Balance, AccountId, DepositBalance> {
+		pub owner: AccountId,
+		pub issuer: AccountId,
+		pub admin: AccountId,
+		pub freezer: AccountId,
+		pub supply: Balance,
+		pub deposit: DepositBalance,
+		pub min_balance: Balance,
+		pub is_sufficient: bool,
+		pub accounts: u32,
+		pub sufficients: u32,
+		pub approvals: u32,
+		pub is_frozen: bool,
+	}
+
+	impl<Balance, AccountId, DepositBalance> AssetDetails<Balance, AccountId, DepositBalance> {
+		pub(super) fn migrate_to_v1(
+			self,
+		) -> super::AssetDetails<Balance, AccountId, DepositBalance> {
+			let status = if self.is_frozen { AssetStatus::Frozen } else { AssetStatus::Live };
+
+			super::AssetDetails {
+				owner: self.owner,
+				issuer: self.issuer,
+				admin: self.admin,
+				freezer: self.freezer,
+				supply: self.supply,
+				deposit: self.deposit,
+				min_balance: self.min_balance,
+				is_sufficient: self.is_sufficient,
+				accounts: self.accounts,
+				sufficients: self.sufficients,
+				approvals: self.approvals,
+				status,
+			}
+		}
+	}
+}
+
+#[frame_support::storage_alias]
+/// Details of an asset.
+pub(crate) type Asset<T: Config<I>, I: 'static> = StorageMap<
+	Pallet<T, I>,
+	Blake2_128Concat,
+	<T as Config<I>>::AssetId,
+	AssetDetails<
+		<T as Config<I>>::Balance,
+		<T as frame_system::Config>::AccountId,
+		DepositBalanceOf<T, I>,
+	>,
+>;
+
+#[derive(PartialEq, Eq, Clone, Encode, Decode, RuntimeDebug, TypeInfo)]
+pub(crate) struct AssetDetails<Balance, AccountId, DepositBalance> {
+	pub owner: AccountId,
+	pub issuer: AccountId,
+	pub admin: AccountId,
+	pub freezer: AccountId,
+	pub supply: Balance,
+	pub deposit: DepositBalance,
+	pub min_balance: Balance,
+	pub is_sufficient: bool,
+	pub accounts: u32,
+	pub sufficients: u32,
+	pub approvals: u32,
+	pub status: AssetStatus,
+}
+
+pub struct MigrateToV1<T, I>(core::marker::PhantomData<(T, I)>);
+impl<T: Config<I>, I: 'static> OnRuntimeUpgrade for MigrateToV1<T, I> {
+	fn on_runtime_upgrade() -> Weight {
+		let in_code_version = Pallet::<T, I>::in_code_storage_version();
+		let on_chain_version = Pallet::<T, I>::on_chain_storage_version();
+		if on_chain_version == 0 && in_code_version == 1 {
+			let mut translated = 0u64;
+			Asset::<T, I>::translate::<
+				old::AssetDetails<T::Balance, T::AccountId, DepositBalanceOf<T, I>>,
+				_,
+			>(|_key, old_value| {
+				translated.saturating_inc();
+				Some(old_value.migrate_to_v1())
+			});
```

Comment:
Hmm is there a bound on the number of Assets that could exist for a runtime? Otherwise we may need to write this as an MBM.

Someone could potentially spam the chain with thousands of dummy Assets right before the migration is scheduled to take place, which would cause the migration block PoV size to be too large and brick the chain.